store/src/main/java/com/zimbra/cs/service/MailboxAttachmentService.java

```diff
@@ -18,7 +19,20 @@ public class MailboxAttachmentService implements AttachmentService {
   public Try<MimePart> getAttachment(
       String accountId, AuthToken token, int messageId, String part) {
     return Try.of(() -> MailboxManager.getInstance().getMailboxByAccountId(accountId))
-        .mapTry(mailbox -> mailbox.getMessageById(new OperationContext(token), messageId))
-        .mapTry(message -> Mime.getMimePart(message.getMimeMessage(), part));
+        .mapTry( mailbox ->
+            mailbox.getMessageById(new OperationContext(token), messageId)
+        )
+        .recoverWith( ex ->
+            Try.failure(ServiceException.NOT_FOUND("File not found.", ex))
```

Comment:
Hi @frisonisland, I suggest to simplify the code with `mapFailure` in order to change the error instead of `recoverWith` + `Try.failure`.

Reply:
@matteobaglini with map failure we get:
```suggestion
            .mapFailure(
            Case(
                $(instanceOf(Exception.class)),
                ex -> ServiceException.NOT_FOUND(("File not found.", ex)));
```
There is also onFailure:
```suggestion
            .onFailure(Exception.class,
                ex -> ServiceException.NOT_FOUND("File not found.", ex))
```
what do you think about it? mapFailure makes more sense, but syntax may be uglier :eyes:

Reply:
My functional brain on Scala/Kotlin votes for onFailure :P

Reply:
Yeah +1 for `onFailure` :-)

Reply:
Thanks to your feedback I found some inconsistencies on the error message

Reply:
To explain: seems like onFailure is ~ "peek" and doesn't alter the result. However after digging in vavr library I found this syntax to be "quite" nice (I  didn't like instanceof):
```
.mapFailure(
                    Case(
                        Pattern0.of(Exception.class),
                        ex -> ServiceException.PARSE_ERROR("Error parsing mime message", ex)
                    )
                )
```